packages/central-server/src/dataAccessors/upsertAnswers.js

```
@@ -2,6 +2,22 @@ import { S3, S3Client, S3_BUCKET_PATH, getS3ImageFilePath } from '@tupaia/server
 import { QuestionType } from '@tupaia/types';
 import { DatabaseError, UploadError } from '@tupaia/utils';
 
+function isValidHttpUrl(str) {
+  try {
+    const url = new URL(str);
+    return url.protocol === 'https:' || url.protocol === 'http:';
+  } catch (e) {
+    if (e instanceof TypeError && e.code === 'ERR_INVALID_URL') {
```

hmmm I am not very familiar with `ERR_INVALID_URL` but a quick search suggests it might not be guaranteed to be part of the error object. I am bit worried that we could get false negatives and miss handling valid data further down. I would probably prefer to just always catch and return false and then let it fail further down in the unlikely event that the data is not a valid url or ID. I'm open to discussion though.

---

Ah, thanks for double checking that!

I’ll keep the `TypeError` check since that’s [explicitly defined in the `URL` standard](https://url.spec.whatwg.org), but will go ahead and remove the `code` check